packages/stream-object/src/index.ts

```diff
@@ -16,15 +29,32 @@ export interface StreamObjectResult<T extends Schema> extends StreamTextResult {
   partialObjectStream: ReadableStream<PartialDeep<Infer<T>>>
 }
 
+export interface StreamObjectOnFinishResult<T extends Schema> {
+  object?: Infer<T>
+}
+
+interface StreamObjectExtraOptions<T extends Schema> {
+  onFinish?: (result: StreamObjectOnFinishResult<T>) => unknown
+}
+
 /** @experimental WIP */
-export const streamObject = async <T extends Schema>(options: StreamObjectOptions<T>): Promise<StreamObjectResult<T>> =>
-  streamText({
+export async function streamObject<T extends Schema>(options: StreamObjectExtraOptions<T> & StreamObjectOptions<T> & { output: 'array' }): Promise<StreamObjectResult<T> & { elementStream: ReadableStream<Infer<T>> }>
```

Comment:
When specifying an `array`, perhaps we should avoid returning a `partialObjectStream`

Reply:
lgtm
let see how to do it